web/components/nav/sidebar.tsx

```diff
@@ -264,7 +260,7 @@ export default function Sidebar(props: { className?: string }) {
           buttonContent={<MoreButton />}
         />
 
-        {user && <CreateQuestionButton user={user} />}
+        {user && !user.isBannedFromPosting && <CreateQuestionButton />}
```

Comment:
One downside of moving the ban condition out is someone might forget to enforce it on another use of the button. But I guess the button isn't really reused much / not a big problem.